src/proxy.rs

```diff
@@ -797,32 +797,39 @@ fn parse_registry_values_impl(registry_values: RegistryProxyValues) -> Result<Sy
             }
         }
     } else {
-        // Use one setting for all protocols.
-        if proxy_server.starts_with("http:") {
-            insert_proxy(&mut proxies, "http", proxy_server);
+        if let Some(scheme) = extract_type_prefix(&proxy_server) {
+            // Explicit protocol has been specified
+            insert_proxy(&mut proxies, scheme, proxy_server);
         } else {
+            // No explicit protocol has been specified, default to HTTP
             insert_proxy(&mut proxies, "http", format!("http://{}", proxy_server));
-            insert_proxy(&mut proxies, "https", format!("https://{}", proxy_server));
+            insert_proxy(&mut proxies, "https", format!("http://{}", proxy_server));
         }
     }
     Ok(proxies)
 }
 
+/// Extract the protocol from the given address, if present
+/// For example, "https://example.com" will return Some("https")
 #[cfg(target_os = "windows")]
-fn contains_type_prefix(address: &str) -> bool {
+fn extract_type_prefix(address: &str) -> Option<String> {
```

Comment:
Slighly refactored the method to also return the schema

Reply:
Perhaps we could return `Option<&str>` here, since the value isn't always used, and then when it is used, the decision to make a copy can be made there.

Reply:
@seanmonstar Good point, thank you. Should be good now 👍
Let me know if you have other concerns :)